mobile_verifier/src/speedtests.rs

```diff
@@ -89,13 +91,21 @@ impl SpeedtestRollingAverage {
             .await;
 
         futures::stream::iter(tests_by_publickey.into_iter())
-            .then(move |(pubkey, cell_speedtests)| async move {
-                let rolling_average = exec
-                    .fetch(&pubkey)
-                    .await?
-                    .unwrap_or_else(|| SpeedtestRollingAverage::new(pubkey.clone()));
-                Ok((rolling_average, cell_speedtests))
+            .then(move |(pubkey, cell_speedtests)| {
```

Comment:
why not just do a`filter` step before this then that eliminiates any that are not valid?

Reply:
The client can return an Error